main/src/hadd.cxx

```diff
@@ -394,7 +394,7 @@ int main( int argc, char **argv )
             while (indirect_file) {
                if( std::getline(indirect_file, line) && line.length() ) {
                   if (gSystem->AccessPathName(line.c_str(), kReadPermission) == kTRUE ||
-                      (!TString(line).EndsWith(".root"))) {
+                      TFile(line.c_str()).IsZombie()) {
```

Comment:
I am unsure about introducing an extra file open operation, it sounds overkill for this objective. Can't we just remove the check for the file extension and see what happens (by introducing a test). The first call to `TFile::Open` that receives an invalid file should show the error `not a ROOT file` .

Reply:
Ok! A test here? https://github.com/root-project/roottest/blob/master/root/io/hadd/test_compression_settings.py

Reply:
Either a new test class in the same file or a new file in the same directory. In the first case, I would say the file needs a renaming because we wouldn't be testing only the compression settings anymore. But in general yes, that sounds like the right place to introduce the new test 👍

Reply:
Done! https://github.com/root-project/roottest/pull/1142